parquet-column/src/main/java/org/apache/parquet/column/statistics/BinaryStatistics.java

```diff
@@ -68,27 +90,23 @@ public void setMinMaxFromBytes(byte[] minBytes, byte[] maxBytes) {
   }
 
   @Override
-  public boolean isSmallerThan(long size) {
-    return !hasNonNullValue() || ((min.length() + max.length()) < size);
+  String toString(Binary value) {
+    // TODO: have separate toString for different logical types?
```

Comment:
Yes, this should be based on the full type and not just assume the value can be converted to string with UTF-8.

Reply:
I think, it would be better to be implemented in a separate issue because there are some open questions and might lead to bigger effort. (E.g. do we want to implement it for statistics only or for Binary? Should it be similar to the comparators to be retrieved from type or implement in the toString? What about the unsigned integers?)
What do you think?

Reply:
Sounds good. Please open a follow-up issue, then.

Reply:
Created PARQUET-1170.